tests/src/io/unitTestTrackingTxtFileReader.cpp

```diff
@@ -0,0 +1,358 @@
+}
+
+BOOST_AUTO_TEST_CASE(TestVikingRangeDataObservationCollection)
+{
+
+  // Load the observations from the Viking file
+  std::shared_ptr<tio::TrackingTxtFileContents> rawVikingFile = readVikingRangeFile(vikingRangePath);
+  auto observationCollection = observation_models::createTrackingTxtFileObservationCollection<double, double>(rawVikingFile, "Viking", {n_way_range});
+
+  // Check size of observations
+  BOOST_CHECK_EQUAL(observationCollection->getTotalObservableSize(), 1258);
+
+  // Check if n-way-range is present in the collection
+  const auto& observationTypeStartSize = observationCollection->getObservationTypeStartAndSize();
+  BOOST_CHECK(observationTypeStartSize.find(n_way_range) != observationTypeStartSize.end());
+
+  //  Checking the first element with station 63 - 63 in the Viking file
+  const LinkDefinition linkDefDsn63({
+                                        {transmitter, LinkEndId("Earth", "DSS-63")},
+                                        {reflector, LinkEndId("Viking", "Antenna")},
+                                        {receiver, LinkEndId("Earth", "DSS-63")},
+                                    });
+
+  auto observationsAndTimesDsn63 = observationCollection->getSingleLinkObservationsAndTimes(n_way_range, linkDefDsn63);
+  auto observationsDsn63 = observationsAndTimesDsn63.first;
+  auto timesDsn63 = observationsAndTimesDsn63.second;
+
+  BOOST_CHECK_CLOSE(observationsDsn63(0, 0), 2371.564782809, 1e-12);
+  BOOST_CHECK_CLOSE(timesDsn63[0], 2442999.2528009 - basic_astrodynamics::JULIAN_DAY_ON_J2000, 1e-8);
+
+
+//  const auto& concatenatedObservations = observationCollection->getObservationVectorReference();
+//  std::vector<double> observationTimes = observationCollection->getConcatenatedTimeVector();
+//  std::cout << observationTimes[0] << "\t" << concatenatedObservations(0, 0) << "\n";
+
+//  std::vector<int> linkEndIds = observationCollection->getConcatenatedLinkEndIds();
+//  std::map<int, observation_models::LinkEnds> linkEndIdMap = observationCollection->getInverseLinkEndIdentifierMap();
+//  for (auto& pair : linkEndIdMap) {
+//    std::cout << pair.first << " - " << pair.second;
+//  }
+}
+
+BOOST_AUTO_TEST_CASE(TestJuiceFile)
+{
+  spice_interface::loadStandardSpiceKernels();
+
+  std::shared_ptr<tio::TrackingTxtFileContents> rawFdetsDopplerFile = readJuiceFdetsFile(juiceFdetsDopplerPath);
+  rawFdetsDopplerFile->addMetaData(tio::TrackingDataType::doppler_base_frequency, 8420.0e6);
+  rawFdetsDopplerFile->addMetaData(tio::TrackingDataType::doppler_bandwidth, 2.0e3);
+  rawFdetsDopplerFile->addMetaData(tio::TrackingDataType::vlbi_station_name, "Hb");
+
+
+  // CHECK THE RAW FILE
+  BOOST_CHECK_EQUAL(rawFdetsDopplerFile->getNumColumns(), 5);
+
+  auto dataMap = rawFdetsDopplerFile->getDoubleDataMap();
+  auto dataBlockLast = extractBlockFromVectorMap(dataMap, -1);
+
+  BOOST_CHECK_EQUAL(dataBlockLast[tio::TrackingDataType::tdb_time_j2000], 735687970.0 + 32.184 + 37);
```

Comment:
This test is failing for me:

/home/dominic/Tudat/tudat-bundle/tudat-bundle/tudat/tests/src/io/unitTestTrackingTxtFileReader.cpp(340): error: in "test_tracking_txt_file_reader/TestJuiceFile": check dataBlockLast[tio::TrackingDataType::tdb_time_j2000] == 735687970.0 + 32.184 + 37.0 has failed [735688039.18554986 != 735688039.18400002]

Reply:
It also fails for me. I was hoping to discuss it with you.

Reply:
Where do you get the "735687970.0 + 32.184 + 37.0" value from, which you use to compare?

Reply:
- De UTC time in dat veld is `2023-04-25T09:46:10.000 `
- Ik heb [nsidc.org](https://nsidc.org/data/icesat/glas-date-conversion-tool/date_convert/) gebruikt om dit om te zetten naar seconden sinds J2000 (735687970).
- 32.184 voor verschil tussen TT en TAI en dan 37 leap seconds.
- Ik dacht eerst dat je die ervan moest aftrekken, maar dit komt heel dicht bij wat de spice functie oplevert (`735688039.18554986 != 735688039.18400002`).
Maar ik begrijp dus niet hoe het nu precies hoort te zijn.

Reply:
I think the issue is in the fact that this is missing the TT->TDB conversion. When I add this conversion, the error goes from 1.5 milliseconds to 5 microseconds. If I understand correctly, the TrackingFileFieldUTCTimeConverter::toDouble converts a UTC time to TDB time (?) I think that this Spice approximation of the UTC->TDB is accurate to about this level, so this checks out.

Reply:
We could use the models for the conversion in SOFA (which Tudat uses for high-accuracy conversions of time/Earth orientation). But, the problem is that at the level of microseconds, the conversion from TT to TDB not only depends on time, but also on location (since the TT<->TDB conversion is a four-dimensional one in which relativistic effects come into play, specifically the fact that time will run 'at a different rate' at different potentials)

Reply:
The following test works:
  double tdbMismatch = dataBlockLast[tio::TrackingDataType::tdb_time_j2000] - ( 735687970.0 + 32.184 + 37.0 ) -
      sofa_interface::getTDBminusTT( dataBlockLast[tio::TrackingDataType::tdb_time_j2000], Eigen::Vector3d::Zero( ) );
  BOOST_CHECK_SMALL(tdbMismatch, 1.0E-5 );

which ensures that the conversion, done using two slightly different ways, is consistent at the 10 microsecond level. For now, I think this is good enough. At some point, we may need to revisit this (in 10 microseconds, the Earth will move about 30 cm w.r.t. the barycenter, which is relevant for high-accuracy applications). But, when we reach a point where we care about this, we'll be doing pretty well :)

Reply:
See section 31 of this: https://dgk.badw.de/fileadmin/user_upload/Files/DGK/docs/a-124.pdf and Chapter 2 of this: https://aa.usno.navy.mil/downloads/Circular_179.pdf about everything you (n)ever wanted to know about time scales :)

Reply:
Thank you! I am still not sure if I did the time conversion correctly for the other times in the toDouble() method. I will definitely check out these sources and let you know!